save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/AvatarStorage.kt

```diff
@@ -51,32 +54,37 @@ class AvatarStorage(configProperties: ConfigProperties) :
      * @return [Path] is built by [AvatarKey] object
      */
     override fun buildPathToContent(rootDir: Path, key: AvatarKey): Path = rootDir
-        .let { if (key.type == AvatarType.USER) it.resolve(USERS_DIRECTORY) else it }
+        .let {
+            when (key.type) {
+                AvatarType.USER -> it.resolve(USERS_DIRECTORY)
+                AvatarType.ORGANIZATION -> it.resolve(ORGANIZATIONS_DIRECTORY)
+                else -> throw IllegalStateException("Not supported type: ${key.type}")
+            }
+        }
         .resolve(key.objectName)
-        .resolve(key.imageName)
 
     companion object {
+        const val ORGANIZATIONS_DIRECTORY = "organizations"
         const val USERS_DIRECTORY = "users"
     }
 }
 
 /**
  * @property type
  * @property objectName
- * @property imageName
  */
 data class AvatarKey(
     val type: AvatarType,
     val objectName: String,
-    val imageName: String,
 ) {
     /**
      * Added for backward compatibility
      *
      * @return relative path to avatar image
      */
     fun getRelativePath(): String = when (type) {
```

Comment:
btw, since now the path is common for all avatars and depends only on organization\user name -- maybe remove `ImageInfo` class and logic related to it

Reply:
Ok. Agree. I'll refactor it in the next pr.